lib/ichabod/resource_set/source_readers/fda_collection_rest_reader.rb

```diff
@@ -0,0 +1,116 @@
+module Ichabod
+  module ResourceSet
+    module SourceReaders
+      class FdaCollectionRestReader < ResourceSet::SourceReader
+        require 'rest-client'
+
+        def read
+          records.collect do |record|
+            ResourceSet::Resource.new(resource_attributes_from_record(record))
+          end
+        end
+
+        private
+        extend Forwardable
+        def_delegators :resource_set, :fda_collection_id
+
+        def resource_attributes_from_record(record)
+          {
+              prefix: resource_set.prefix,
+              available: available_link_from_record(record),
+              identifier: dc_attribute_from_record('identifier', record),
+              title: dc_attribute_from_record('title', record),
+              creator: dc_attribute_from_record('creator', record),
+              publisher: dc_attribute_from_record('publisher', record),
+              type: dc_attribute_from_record('type', record),
+              description: dc_attribute_from_record('description', record),
+              date: dc_attribute_from_record('date.issued', record),
+              format: dc_attribute_from_record('format', record),
+              language: dc_attribute_from_record('language', record),
+              relation: dc_attribute_from_record('relation', record),
+              rights: dc_attribute_from_record('rights', record),
+              subject: dc_attribute_from_record('subject', record)
+          }
+        end
+
+        def dc_attribute_from_record(attribute, record)
+          values = []
+          record['metadata'].each do |element|
+            if element['key'].include? "dc.#{attribute}"
+              values << element['value']
+            end
+          end
+          values
+        end
+
+        def available_link_from_record(record)
+          values = []
+          record['metadata'].each do |element|
+            if element['key'].include? "dc.identifier"
```

Comment:
@sgbalogh ,

Could you make this a one line conditional statement? 
```
if element['key'].include? "dc.identifier" && element['value'].include? "handle"
        values << element['value']
    end
```
 unless I'm missing something?